docs/ReleaseHistory_Rust.md

```diff
@@ -0,0 +1,27 @@
+# Release Notes
+This document contains release notes pertaining to the Rust crates.
+
+## Definitions
+
+- RUL => New detection.
+- DEP => Update dependency.
+- BRK => General breaking change.
+- BUG => General bug fix.
+- NEW => New API or feature.
+- PRF => Performance work.
+- FPS => False positive reduction in static analysis.
+- FNS => False negative reduction in static analysis.
+
+# UNRELEASED
+- DEP: `System.Text.Json` updated to `v8.0.4` to resolve Depandabot alert.
+- NEW: Introduces the `ScanEngine` struct, which now encapsulates the internal state tracking previously managed by the `Scan` struct. This refactor simplifies the usage of the `Scan` struct in scoped thread contexts, removing the requirement for a `Mutex` or `Arc`.
```

Comment:
You have this kind of backwards :)

The ScanEngine is the base, that is entirely immutable and takes a single mutable ScanState. This makes it possible for concurrent usage.

The Scan struct is unchanged from the customer perspective, however, it now leverages an internal ScanState and ScanEngine to ensure no-breaking changes.

Reply:
Customers wishing to use concurrency without any locking should use a single ScanEngine instance with ScanState's per-invocation/thread.

Reply:
@beaubelgrave I think I understand your point. Took another stab at authoring the note.

Reply:
> This refactor simplifies the usage of the `Scan` struct in scoped thread contexts, removing the requirement for a `Mutex` or `Arc`.

This should change to:
This refactor allows the usage of a single `ScanEngine` struct in scoped thread contexts, removing the requirement for a `Mutex` or `Arc`.

Reply:
@beaubelgrave I think you replied on an outdated version of the note. I updated it to: 

Introduces the `ScanEngine` struct, which allows simplified usage in concurrent scenarios---a single `ScanEngine` instance, along with per-thread `ScanState` instances, suffice without the need for additional synchronization. The existing `Scan` struct is operationally unchanged for users.

Reply:
Perfect! Let's wait for a release until we get the Rc's removed, I think that's the last step to full concurrency.

Reply:
Sounds good! Thanks @beaubelgrave .